Fixed clear-app-data exception

IlyaBausovAkvelon opened the pull request:
In this fix:
- Updated idb version to 1.1.7 and fixed issues after that.
- Fixed uncatched clear-app-data exception with that.
- Added Integration and Unit tests to cover clear-app-data command.

IlyaBausovAkvelon committed:
Updated idb to 1.1.7.

IlyaBausovAkvelon committed:
Added missed catch for clear-app-data exception.

IlyaBausovAkvelon committed:
Improved checking for if application is installed.

Checking for if application is installed lead to failing with error saying "error checking if application is installed". Even in cases when application is not installed.

This should be just leading to NO result, not to the app error.

IlyaBausovAkvelon committed:
Added test for simulator to check clear-app-data command.

IlyaBausovAkvelon committed:
Updated make script to build frameworks under arm64.

Updated script and rebuilt frameworks.

AlexeyBudynkov commented:
At this moment these command doesn't work on M1:
```
$ make unit-tests
$ make integration-tests
```

_Testing failed:
	The linked and embedded framework 'CocoaLumberjack.framework' is missing one or more architectures required by this target: arm64.
	The linked and embedded framework 'OCMock.framework' is missing one or more architectures required by this target: arm64.
	The linked and embedded framework 'Specta.framework' is missing one or more architectures required by this target: arm64.
	The linked and embedded framework 'Expecta.framework' is missing one or more architectures required by this target: arm64.
	The linked and embedded framework 'CocoaLumberjack.framework' is missing one or more architectures required by this target: arm64.
	Testing cancelled because the build failed._

This command creates other error:
`$ make cli-tests`

_An error occurred while loading ./spec/upload_xctestconfig_spec.rb.
Failure/Error: if RunLoop::Version.new(xcode_version) >= min_xcode_version_

_NoMethodError:
  undefined method `strip' for nil:NilClass_

AlexeyBudynkov commented on iOSDeviceManager/Devices/PhysicalDevice.m:
```diff
@@ -357,10 +357,12 @@ - (iOSReturnStatusCode)isInstalled:(NSString *)bundleID {
-        @throw [NSException exceptionWithName:@"IsInstalledAppException"
-                                       reason:@"Unable to determine if application is installed"
-                                     userInfo:nil];
+        if ([[error localizedDescription] containsString:@"is not installed."]) {
```
What if we'll have [error localizedDescription] result in other language than English?

IlyaBausovAkvelon replied:
Fixed.

IlyaBausovAkvelon committed:
Updated Expecta, Specta and OCMock XCFrameworks.

IlyaBausovAkvelon committed:
Updated CocoaLumberjack in Vendor directory.

IlyaBausovAkvelon committed:
Updated CocoaLumberjack in Frameworks directory.

IlyaBausovAkvelon committed:
Updated CocoaLumberjack usage in project.

Since we decided to move forward from several forks in our projects we are removing CocoaLumberjack fork usage in iOSDeviceManager.
I've moved all IDM related logging settings to IDMLogger class and fixed deprecations.

IlyaBausovAkvelon committed:
Removed testSimulatorAppURL test since removing of method which is being tested in it.

IlyaBausovAkvelon committed:
Added IDMLogger to Unit and Integration targets.

Commit:
1) returned state function to make this value mocked
2) fixed usage of ConsoleWriteErr
3) incremented a minor version to make tests launch on a simulator (without it a debugger referenced to a previous non-existing version of iOS)
4) tests were remade to use a boot method
5) result of the tests testWaitForBootableStateBootingWithStateUnknown, testWaitForBootableStateWithStateCreatingFailure now returns true, because some logic changed

IlyaBausovAkvelon committed:
Replaced OCMExpect with OCMStub in SimulatorTest.

IlyaBausovAkvelon committed:
Refactored Simulator's boot method.

IlyaBausovAkvelon committed:
Fixed failing unit tests.

AlexeyBudynkov commented on iOSDeviceManager/Utilities/DeviceUtils.m:
```diff
@@ -150,7 +150,7 @@ + (NSString *)defaultSimulatorID {
 + (NSString *)defaultSimulator {
     NSUInteger major = XcodeUtils.versionMajor + 2;
-    NSUInteger minor = XcodeUtils.versionMinor;
+    NSUInteger minor = XcodeUtils.versionMinor+1;
```
I'm not sure about this workaround.

IlyaBausovAkvelon committed:
Added default device mapping for Xcode 13.x.

IlyaBausovAkvelon committed:
Fixed checking for installed apps in PhysicalDevice.

IlyaBausovAkvelon committed:
Update azure-pipelines.yml

Commit:
unit test fix

IlyaBausovAkvelon committed:
Updated CocoaLumberjack framework linking process.

IlyaBausovAkvelon committed:
Merge branch 'fix/clear-app-data-exception' of https://github.com/calabash/iOSDeviceManager into fix/clear-app-data-exception

IlyaBausovAkvelon committed:
Updated CocoaLumberjack settings in project file.

MatkovIvan commented on IDMLogger.h:
```diff
@@ -0,0 +1,19 @@
+//
```
I do not think that it should be placed at the root of repo

MatkovIvan commented on IDMLogger.m:
```diff
@@ -0,0 +1,230 @@
+//
```
Same

MatkovIvan commented on IDMLogger.m:
```diff
@@ -0,0 +1,230 @@
+  NSString *logLevel;
+  switch (logMessage.flag) {
+    case DDLogFlagError    : logLevel = @"ERROR"; break;
+    case DDLogFlagWarning  : logLevel = @" WARN"; break;
```
Extra space in string

MatkovIvan commented on iOSDeviceManager/Devices/Simulator.m:
```diff
@@ -159,14 +160,18 @@ - (BOOL)waitForSimulatorState:(FBiOSTargetState)state
 
+- (FBiOSTargetState)state {
+    return self.fbSimulator.state;
+};
```
```suggestion
}
```

IlyaBausovAkvelon committed:
Fixed testWaitForBootableStateWithStateCreatingSuccess test in SimulatorTest.

IlyaBausovAkvelon committed:
Fixed formatting in IDMLogger and Simulator classes.

IlyaBausovAkvelon committed:
Moved IDMLogger to the Utilities.
